src/fiber/gossip.rs

```diff
@@ -2293,28 +2230,16 @@ where
         state: &mut Self::State,
     ) -> Result<(), ActorProcessingErr> {
         match message {
-            GossipActorMessage::PeerConnected(peer_id, pubkey, session) => {
+            GossipActorMessage::PeerConnected(peer_id, _pubkey, session) => {
                 if state.is_peer_connected(&peer_id) {
-                    warn!(
-                        "Repeated connection from {:?} for gossip protocol",
-                        &peer_id
-                    );
                     return Ok(());
                 }
-                debug!(
-                    "Gossip peer connected: peer {:?}, pubkey {:?}, session {:?}",
-                    &peer_id, &pubkey, &session.id
-                );
                 state
                     .peer_states
                     .insert(peer_id.clone(), PeerState::new(session.id, session.ty));
             }
-            GossipActorMessage::PeerDisconnected(peer_id, session) => {
-                debug!(
-                    "Peer disconnected: peer {:?}, session {:?}",
-                    &peer_id, &session.id
-                );
-                drop(state.peer_states.remove(&peer_id));
```

Comment:
The drop here is useful, as that would stop the filter processor actor.

Reply:
why we need to explicitly call drop to release data removed from a HashMap?

Reply:
This is used to drop the value saved in the HashMap. The value of type `PeerState` https://github.com/contrun/fiber/blob/b65b586f27ea79d6ea0f1cd677b64b693e12ce60/src/fiber/gossip.rs#L778-L805 maintains a few things.

Reply:
I don't get you, letting the removed value go out of scope is sufficient, and rust will handle memory cleanup automatically, could you please tell why this change will prevent the removed `peerState` from dropping?
```diff
-                drop(state.peer_states.remove(&peer_id));
+                state.peer_states.remove(&peer_id); 
```

Reply:
Sorry. I missed the following line. You're right.